packages/curve-ui-kit/src/shared/ui/Metric.tsx

```diff
@@ -0,0 +1,233 @@
+    setOpenCopyAlert(true)
+  }
+
+  return (
+    <Box display="flex" flexDirection="column" alignItems={alignment} gap={Spacing.xs}>
+      <Typography variant="bodyXsRegular" color="textTertiary">
+        {label}
+        {tooltip && (
+          <Tooltip arrow placement="top" title={tooltip}>
+            <span>
+              &nbsp;
+              <InfoOutlinedIcon sx={{ fontSize: '1.25em' }} />
+            </span>
+          </Tooltip>
+        )}
+      </Typography>
+
+      {loading ? (
+        <Skeleton
+          variant="text"
+          width="100%"
+          sx={{
+            fontSize: TYPOGRAPHY_VARIANTS[MetricSize[size]],
+          }}
+        />
+      ) : (
+        <Box display="flex" gap={Spacing.xs} alignItems="baseline">
+          <Tooltip
+            arrow
+            placement="bottom"
+            title={value.toLocaleString()}
+            onClick={copyValue}
+            sx={{
+              cursor: 'pointer',
+            }}
+          >
+            <Box display="flex" gap={Spacing.xxs} alignItems="baseline">
+              {unit?.position === 'prefix' && (
+                <Typography variant={MetricSize[size]} color="textSecondary">
+                  {unit.symbol}
+                </Typography>
+              )}
+
+              <Typography variant={MetricSize[size]} color="textPrimary">
+                {formatter(abbreviate ? abbreviateNumber(value) : value)}
+              </Typography>
+
+              {abbreviate && (
+                <Typography variant={MetricSize[size]} color="textPrimary" textTransform="capitalize">
+                  {scaleSuffix(value)}
+                </Typography>
+              )}
+
+              {unit?.position === 'suffix' && (
+                <Typography variant={MetricSize[size]} color="textSecondary">
+                  {unit.symbol}
+                </Typography>
+              )}
+            </Box>
+          </Tooltip>
+
+          {(change || change === 0) && (
+            <Typography
+              variant="highlightM"
+              color={change > 0 ? 'success' : change < 0 ? 'error' : 'textHighlight'}
+              sx={{ marginInlineStart: Spacing.xs }} // Not part of Figma, but my own addition
+            >
+              {formatChange(change)}%
+            </Typography>
+          )}
+        </Box>
+      )}
+
+      {notional !== undefined && (
+        <Typography variant="highlightXsNotional" color="textTertiary">
+          {notionalUnit?.position === 'prefix' && notionalUnit.symbol}
+          {notionalFormatter(notionalAbbreviate ? abbreviateNumber(notional) : notional)}
+          {notionalAbbreviate && scaleSuffix(notional)}
+          {notionalUnit?.position === 'suffix' && notionalUnit.symbol}
+        </Typography>
+      )}
+
+      <Snackbar open={openCopyAlert} autoHideDuration={6000} onClose={() => setOpenCopyAlert(false)}>
+        <Alert variant="filled" severity="success">
+          Copied metric value: {value}
```

Comment:
I just noticed the font family seems broken here, because we don't use the global `CssBaseline` (as that could conflict with the existing code). We might need to wrap this in `Typography` or configure the `Alert`'s `styleOverrides`

Reply:
btw, do we need to add the actual value to the snackbar?

Reply:
Debatable, I figured it's nice as an extra confirmation to see what you've just copied to the clipboard. Might be information overload, I don't have a strong opinion on this matter. @0xtutti ?